Feature :  Search Todo

ich-wtag opened the pull request:
**Overview**

[Task-5: Search Todo](https://trello.com/c/4wrjnUOP/5-task-5-search-todo)
This PR includes **Todo Search** functionality. Here is an input button with a **Search button**. User can search the desired todo items in the list by using **Todo's Title** . If the input box is empty and user click's the Search Button, then whole Todo's 
 
**Types of Changes**

- [x] There is a new input element on the top of document with a button along with it.
- [x] When a search phrase is entered and search button is clicked, relevant todos are shown. 

Review
Please have a close look on this PR and kindly put your valuable comments.


[search_todo.webm](https://github.com/ich-wtag/Todo-App_Vanilla_Js/assets/153156568/32f1f613-79a8-4721-8a9c-21f1acc4827d)

ich-wtag committed:
Feat : Search todo functionality added and isComplet property refactored in addTodoHandler

ich-wtag committed:
Merge branch 'Feature__Mark-Done-Todo' of https://github.com/ich-wtag/Todo-App_Vanilla_Js into Feature__Search-Todo

ich-wtag committed:
Refactor: code refactored to see the done todo items marked while searching

ich-wtag committed:
Refactor: code refactored in searchHandler function to remove the unncessary code and relocated to maintain the sequence of work

hhu-wtag commented on script.js:
```diff
@@ -22,6 +24,7 @@ const addTodoHandler = () => {
         id: new Date().getTime(),
         title: todoTitle,
         isEditing: false,
+        isCompleted: false,
```
This change is not supposed to be in this PR.

hhu-wtag commented on script.js:
```diff
@@ -109,7 +111,15 @@ const markDoneTodoHandler = (
+};
+
+const searchHandler = () => {
+    let searchedArray = todos.filter((todo) =>
```
```suggestion
    const searchedArray = todos.filter((todo) =>
```

hhu-wtag commented on script.js:
```diff
@@ -132,6 +142,12 @@ const createTodoElement = (todo) => {
+        $paragraphElement.classList.add("done-todo");
+        $doneButton.classList.add("hide");
+        $editButton.classList.add("hide");
+    }
```
ditto

ich-wtag committed:
Refactor : code refactored in searchHandler to make it lowercase and trim the searched input values.

ich-wtag committed:
Merge branch 'Feature__Mark-Done-Todo' of https://github.com/ich-wtag/Todo-App_Vanilla_Js into Feature__Search-Todo

ich-wtag committed:
reafctor : code refactored to make the search functionality properly

nmi-wtag commented on script.js:
```diff
@@ -112,6 +115,16 @@ const markDoneTodoHandler = (
 };
 
+const searchHandler = () => {
+    let searchedValue = $searchInput.value.toLowerCase().trim();
```
```suggestion
    const searchedValue = $searchInput.value.toLowerCase().trim();
```

nmi-wtag commented on script.js:
```diff
@@ -185,3 +204,4 @@ const renderTodos = (todos) => {
 };
 
 $addButton.addEventListener("click", addTodoHandler);
+$searchButton.addEventListener("click", searchHandler);
```
```suggestion
$searchInput.addEventListener("keyup", searchHandler);
```

ich-wtag committed:
Refactor : code refactor to clear the search input field when adding, deleting and editing the todos

ich-wtag committed:
Update script.js

Refactor : code refactored to make it more effective way to clear the search input field using existing a utility function.

hhu-wtag reviewed (approved):
Approved

ich-wtag committed:
refactor : resolve the review

code refactor to resolve the review and make a new showCompletedTodo added to make visualize the completed todos.

ich-wtag committed:
Merge branch 'main' of https://github.com/ich-wtag/Todo-App_Vanilla_Js into Feature__Search-Todo
